File: src/types/message.ts
// Message types - aligned with Rust backend (ARCHITECTURE.md)
export interface Message {
  id: number;
  channel_id: number;
  user_id: number;
  content: string;
  attachments?: string | null;
  created_at: string;
  updated_at: string;
}

export interface CreateMessage {
  channel_id: number;
  user_id: number;
  content: string;
  attachments?: string | null;
}

export interface Channel {
  id: number;
  name: string;
  description?: string;
  unread_count?: number;
}

File: src-tauri/src/models/message.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Message {
    pub id: i64,
    pub channel_id: i64,
    pub user_id: i64,
    pub content: String,
    pub attachments: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateMessage {
    pub channel_id: i64,
    pub user_id: i64,
    pub content: String,
    pub attachments: Option<String>,
}

